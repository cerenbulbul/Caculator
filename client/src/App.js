
import { useState, useEffect } from 'react'
import io from "socket.io-client";

const socket = io("http://localhost:3001/");


function App() {

  const [getCalc, setCalc] = useState("");
  const [getResult, setResult] = useState("");
  const [getHistorry, setHistory] = useState([]);

  const ops = ['/', '*', '+', '-', '.'];

  useEffect(() => {
    getHistoryData();
  }, []);

  const getHistoryData = () => {
    socket.on('output-history', data => {
      const result = [];
      if (data.length) {
        data.forEach(calc => {
          result.push(calc.calculator)
        });
      }
      result.reverse();
      setHistory(result)
    })
  }

  const updateCalculate = value => {
    if (ops.includes(value) && getCalc === '' ||
      ops.includes(value) && ops.includes(getCalc.slice(-1))) {
      return;
    }

    setCalc(getCalc + value)

    if (!ops.includes(value)) {
      setResult(eval(getCalc + value).toString());
    }
  }

  const showCalculate = () => {
    setCalc(eval(getCalc).toString())
    socket.emit('history', eval(getCalc).toString())
    window.location.reload();
    //setHistory([eval(getCalc).toString(), getHistorry])
  }

  const deleteLast = () => {
    if (getCalc === '') {
      return;
    }
    const value = getCalc.slice(0, -1);
    setCalc(value)
  }


  const createDigits = () => {
    const digits = [];

    for (let i = 1; i < 10; i++) {
      digits.push(
        <button key={i} onClick={() => updateCalculate(i.toString())}>{i}</button>
      )
    }
    return digits;
  }


  const showHistory = () => {
    const showhistory = [];
    let lenght = 0;

    if (getHistorry.length < 10) {
      lenght = getHistorry.length
    }
    else {
      lenght = 10;
    }
    for (let i = 0; i < lenght; i++) {
      showhistory.push(
        <pre className="history-items" key={i}>{getHistorry[i]}</pre>
      )
    }
    return showhistory;
  }

  return (
    <div className="App">
      <div className="calculator">
        <div className="display">
          {getResult ? <span>({getResult})</span> : ''}&nbsp;
          {getCalc || "0"}
        </div>

        <div className="operators">
          <button onClick={() => updateCalculate('/')}>/</button>
          <button onClick={() => updateCalculate('*')}>*</button>
          <button onClick={() => updateCalculate('+')}>+</button>
          <button onClick={() => updateCalculate('-')}>-</button>

          <button onClick={() => deleteLast()}>DEL</button>
        </div>

        <div className="digits">
          {createDigits()}
          <button onClick={() => updateCalculate('0')}>0</button>
          <button onClick={() => updateCalculate('.')}>.</button>
          <button onClick={() => { showCalculate(); }}>=</button>
        </div>
      </div>

      <div className="history">
        <h2>History</h2>

        <div className="history-show">
          {showHistory()}
        </div>

      </div>
    
    </div>
  );
}

export default App;
