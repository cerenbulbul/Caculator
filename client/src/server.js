const express = require('express')
mongoose = require('mongoose')
const cors = require('cors')

const app = express();
app.use(express.json());
app.use(cors());

