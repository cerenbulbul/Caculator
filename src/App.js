
import { useState, useEffect } from 'react'

function App() {

  const [getCalc, setCalc] = useState("");
  const [getResult, setResult] = useState("");
  const [getHistorry, setHistory] = useState([]);

  const ops = ['/', '*', '+', '-', '.'];

  const updateCalculate = value => {
    if (ops.includes(value) && getCalc === '' ||
      ops.includes(value) && ops.includes(getCalc.slice(-1))) {
      return;
    }

    setCalc(getCalc + value)

    if (!ops.includes(value)) {
      setResult(eval(getCalc + value).toString());
    }
  }

  const showCalculate = () => {
    setCalc(eval(getCalc).toString())
    
  }

  const deleteLast = () => {
    if (getCalc === '') {
      return;
    }
    const value = getCalc.slice(0, -1);

    setCalc(value)
  }


  const createDigits = () => {
    const digits = [];

    for (let i = 1; i < 10; i++) {
      digits.push(
        <button key={i} onClick={() => updateCalculate(i.toString())}>{i}</button>
      )
    }
    return digits;
  }


  const showHistory = () => {
    const digits = [];

    for (let i = 0; i < 10; i++) {
      digits.push(
        <p>{i}</p>
      )
    }
    return getHistorry;
  }

  return (
    <div className="App">
      <div className="calculator">
        <div className="display">
          {getResult ? <span>({getResult})</span> : ''}&nbsp;
          {getCalc || "0"}
        </div>

        <div className="operators">
          <button onClick={() => updateCalculate('/')}>/</button>
          <button onClick={() => updateCalculate('*')}>*</button>
          <button onClick={() => updateCalculate('+')}>+</button>
          <button onClick={() => updateCalculate('-')}>-</button>

          <button onClick={() => deleteLast()}>DEL</button>
        </div>

        <div className="digits">
          {createDigits()}
          <button onClick={() => updateCalculate('0')}>0</button>
          <button onClick={() => updateCalculate('.')}>.</button>
          <button onClick={() => showCalculate()}>=</button>
        </div>
      </div>

      <div className="history">
        <h2>History</h2>

        <div className="history-show">
          {showHistory()}
        </div>

      </div>
    </div>
  );
}

export default App;
