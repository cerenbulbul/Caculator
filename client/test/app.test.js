
import { ReactDOM } from 'react-dom'
import React from 'react';
import 'regenerator-runtime/runtime'
import { cleanup, fireEvent, render } from '@testing-library/react';
import { create } from "react-test-renderer";
import App from '../src/App'
import Enzyme,{ shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import * as actions from '../src/App'

Enzyme.configure({ adapter: new Adapter() });

describe("<App />", () => {
  it("Renders <App /> component correctly", async () => {
    const wrapper = shallow(<App/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  test("Button", () => {
    const wrapper = shallow(<App/>)
    expect(wrapper.find('button').length).toBe(17)
  });

  test("div", () => {
    const wrapper = shallow(<App/>)
    expect(wrapper.find('div').length).toBe(7)
  });

  test("h2", () => {
    const wrapper = shallow(<App/>)
    expect(wrapper.find('h2').length).toBe(1)
  });
  
});

