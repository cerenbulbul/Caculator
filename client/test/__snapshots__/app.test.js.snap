// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Matches the snapshot 1`] = `<button />`;

exports[`<App /> Renders <App /> component correctly 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      Â 
      0
    </div>
    <div
      className="operators"
    >
      <button
        onClick={[Function]}
      >
        /
      </button>
      <button
        onClick={[Function]}
      >
        *
      </button>
      <button
        onClick={[Function]}
      >
        +
      </button>
      <button
        onClick={[Function]}
      >
        -
      </button>
      <button
        onClick={[Function]}
      >
        DEL
      </button>
    </div>
    <div
      className="digits"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        1
      </button>
      <button
        key="2"
        onClick={[Function]}
      >
        2
      </button>
      <button
        key="3"
        onClick={[Function]}
      >
        3
      </button>
      <button
        key="4"
        onClick={[Function]}
      >
        4
      </button>
      <button
        key="5"
        onClick={[Function]}
      >
        5
      </button>
      <button
        key="6"
        onClick={[Function]}
      >
        6
      </button>
      <button
        key="7"
        onClick={[Function]}
      >
        7
      </button>
      <button
        key="8"
        onClick={[Function]}
      >
        8
      </button>
      <button
        key="9"
        onClick={[Function]}
      >
        9
      </button>
      <button
        onClick={[Function]}
      >
        0
      </button>
      <button
        onClick={[Function]}
      >
        .
      </button>
      <button
        onClick={[Function]}
      >
        =
      </button>
    </div>
  </div>
  <div
    className="history"
  >
    <h2>
      History
    </h2>
    <div
      className="history-show"
    />
  </div>
</div>
`;
